src/views/ApartmentApp/components/StaffMenu/index.js

```diff
@@ -46,7 +46,7 @@ const StaffMenu = ({ userProfile, authentication }) => {
     setLoading(true);
     let applicationDetailsArray = await housing.getAllApartmentApplications();
     if (applicationDetailsArray) {
-      setApplications(applicationDetailsArray);
+      setApplications(applicationDetailsArray ?? []);
     }
```

Comment:
This is fairly redundant, because `null` is falsy in JavaScript, so the `if` case ensures that `setApplications(...)` will never get called if `applicationDetailsArray` is `null`. Perhaps you meant to remove the `if` so that it will be set to an empty array if `applicationDetailsArray` is falsy?

Reply:
I think I did mean to remove the `if`

I think I realized it probably should just be in a try-catch or using the `.then` stuff, since the null case is just the case where there was an error, including the case where there are no applications at all in the database

I also don't think there is any reason to put this in the useCallback instead of just putting it directly in the useEffect, unless we plan to add a button to call that function, which seems silly if it is mostly equivalent to just refreshing the page

It seems like both the `loadAllCurrentApplications` and the `generateCSVData` might be better to just put inside one `useEffect`